### PR TITLE
http method testing for GET,POST,OUT,HEAD

### PR DESCRIPTION
**Pull request by ilyaanisimov-kaazing**
rfc 7231 tests

### CODE REVIEW

File: specification/http/src/test/java/org/kaazing/specification/http/rfc7231/RequestMethodsIT.java [hunk 1]
```diff
@@ -15,39 +15,78 @@
...
+        k3po.finish();
     }
 
     @Test
-    @Ignore("Not Implemented")
-    public void serverShouldRespondWith405ToUnrecognizedMethods() {
-        // 405 (Method Not Allowed)
+    @Specification({
+        "405/request",
+        "405/response" })
```
Comment (dpwspoon) [inline specification/http/src/test/java/org/kaazing/specification/http/rfc7231/RequestMethodsIT.java]: We should rename this test to more properly reflect the RFC, i.e change Unrecognized to "NotAllowed" or something like that.

>When a request method is received that is known by an origin
   server but not allowed for the target resource, the origin server
   SHOULD respond with the 405 (Method Not Allowed) status code.

File: specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/method.definitions/post/response.rpt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+accepted
+connected
+
+read method "POST"
+read header "Content-Length" "7"
+read "content"
+read closed
+	
+write status "200" "OK"
+write header content-length
```
Comment (dpwspoon) [inline specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/method.definitions/post/response.rpt]: I don't think you need the `write header content-length` for most of these response scripts.  They don't hurt but they don't really add anything IMO

**Comment by dpwspoon** [comment]
LGTM (Looks good to me)

File: specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/method.definitions/post/response.rpt (hunk 2 above)
Comment (ilyaanisimov-kaazing) [inline specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/method.definitions/post/response.rpt]: Script fails if I don't have write header content-length. Let's discuss

**Comment by ilyaanisimov-kaazing** [comment]
Ping @dpwspoon

**Pull request merged**